src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQueryPdao.java

```diff
@@ -92,4 +100,86 @@ public static boolean tooManyDmlStatementsOutstanding(PdaoException ex) {
         && (ex.getCause().getMessage().contains("Too many DML statements outstanding against table")
             || ((BigQueryException) ex.getCause()).getReason().contains("jobRateLimitExceeded"));
   }
+
+  // VIEW DATA
+  public static final String DATA_TEMPLATE = "SELECT <columns> FROM <table> <filterParams>";
+
+  public static final String DATA_FILTER_TEMPLATE =
+      "<whereClause> ORDER BY <sort> <direction> LIMIT <limit> OFFSET <offset>";
+
+  /*
+   * WARNING: Ensure input parameters are validated before executing this method!
+   */
+  public static List<Map<String, Object>> getTable(
+      FSContainerInterface tdrResource,
+      String bqFormattedTableName,
+      List<String> columnNames,
+      int limit,
+      int offset,
+      String sort,
+      SqlSortDirection direction,
+      String filter)
+      throws InterruptedException {
+    final BigQueryProject bigQueryProject = BigQueryProject.from(tdrResource);
+    final String datasetProjectId = bigQueryProject.getProjectId();
+    String whereClause = StringUtils.isNotEmpty(filter) ? filter : "";
+
+    String columns = String.join(",", columnNames);
+    // Parse before querying because the where clause is user-provided
+    final String sql =
```

Comment:
This might want to be in shared code since this is part of the request validation and is needed for both azure and GCP

Reply:
Ah, that's a great point. Thanks for pointing that out. I think I will leave this for the Azure PR so I can take a broader look at where I want to put that shared code.